### PR TITLE
Implement parts of the list API

### PR DESCRIPTION
**Pull request by Rudloff**
Hello,

This PR implements the following API endpoints (see #4134):
* lists/statuses
* lists/destroy
* lists/update
* lists/create
* lists/ownerships

### CODE REVIEW

File: include/api.php [hunk 1]
```diff
@@ -3269,38 +3269,174 @@ function api_help_test($type)
...
+
+	if (api_user() === false) {
+		throw new ForbiddenException();
+	}
+
+	// params
+	$user_info = api_get_user($a);
+	$uid = $user_info['uid'];
+
+	$r = q(
```
Comment (MrPetovan) [inline include/api.php]: Please use `dba` methods for new queries. In this case, a `dba::select` would be more appropriate.

Comment (MrPetovan) [reply include/api.php]: Additionally, please use meaningful names for query result variable names.

**Comment by Rudloff** [comment]
@MrPetovan I replaced `q()` with dba functions everywhere except for the big query in `api_lists_statuses()` because I'm not sure it can be done with `dba::select()`.
(This query is borrowed from `api_statuses_home_timeline()`.)

**Comment by Rudloff** [comment]
@MrPetovan Thanks for the tips! I made some changes.

File: include/api.php [hunk 2]
```diff
@@ -3354,7 +3354,9 @@ function api_lists_statuses($type)
 	unset($_GET["screen_name"]);
 
 	$user_info = api_get_user($a);
-	// get last newtork messages
+	if (!x($_REQUEST, 'list_id')) {
```
Comment (MrPetovan) [inline include/api.php]: Please use `empty()` instead of `!x()`, `x()` will be deprecated in upcoming version.

File: include/api.php [hunk 3]
```diff
@@ -3390,18 +3391,15 @@ function api_lists_statuses($type)
 		STRAIGHT_JOIN `contact` ON `contact`.`id` = `item`.`contact-id` AND `contact`.`uid` = `item`.`uid`
 			AND (NOT `contact`.`blocked` OR `contact`.`pending`)
 		STRAIGHT_JOIN `group_member` ON `group_member`.`contact-id` = `item`.`contact-id`
-		WHERE `item`.`uid` = %d AND `verb` = '%s'
+		WHERE `item`.`uid` = ? AND `verb` = ?
 		AND `item`.`visible` AND NOT `item`.`moderated` AND NOT `item`.`deleted`
-		$sql_extra
```
Comment (MrPetovan) [inline include/api.php]: You still need to concatenate `$sql_extra` in your query.

File: include/api.php [hunk 4]
```diff
@@ -3269,38 +3269,155 @@ function api_help_test($type)
...
+		AND `item`.`id`>?
+		AND `group_member`.`gid` = ?
+		ORDER BY `item`.`id` DESC LIMIT ".intval($start)." ,".intval($count),
+		api_user(),
+		ACTIVITY_POST,
+		$since_id,
+		$_REQUEST['list_id']
+	);
+
+	$items = api_format_items($statuses, $user_info, false, $type);
```
Comment (annando) [inline include/api.php]: The first parameter needs to be an array, here it is an object. Please wrap it in ```dba::inArray($statuses);```

Comment (MrPetovan) [reply include/api.php]: Ah, dang, I thought about checking for that and then I forgot.

Comment (Rudloff) [reply include/api.php]: It still works like this since `foreach` totally works on PDOStatement objects, but I will update my branch.

Comment (MrPetovan) [reply include/api.php]: Really? I have to check this.

Comment (MrPetovan) [reply include/api.php]: Indeed, but it would still bypass calling `dba::fetch` it seems.

**Pull request merged**